server/server.js

```diff
@@ -120,21 +144,76 @@ io.on('connection', function(socket){
   });
 
 
-  // Tower requests all drones for an update every 4 seconds
+  socket.on("DT_updateAck",function(msg){
+    dSay(msg.transcript);
+    tSay("Tower now updating flight path in database");
+
+    // store route into the database
+    var pathUpdate = pendingPathUpdates[msg.callSign];
+
+    utils.getFlightData(msg.callSign)
+    .then(function(pathData){
+      var originalPath = JSON.parse(pathData[0].path_geom);
+      var newPath = originalPath.coordinates.slice(0,pathUpdate.timeBufPrevPtInd).concat(pathUpdate.path);
+      var newPathGeoJSON = JSON.stringify({"type":"LineString","coordinates":newPath});
+      return utils.updateFlightPath(msg.callSign, newPathGeoJSON)
+    })
+    .then(function(result){
+      console.log('updated database')
+      console.log(result)
+      delete pendingPathUpdates[msg.callSign];
```

Comment:
deletes the path update from pendingPathUpdates after the drone acknowledges it updated its path